testing/regress/ecl/nestedjoin.ecl

```diff
@@ -0,0 +1,90 @@
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+// IMPORTANT: this test is generated by the perl script genjointest.pl, so do not edit it by hand
+
+#option('convertJoinToLookup', 0);
+#option('noAllToLookupConversion', 1);
+#option('spanMultipleCpp', 1);
+
+childrec := RECORD,MAXLENGTH(100)
+    UNSIGNED1 i;
+    STRING3 lstr;
+    STRING3 rstr;
+    UNSIGNED1 c;
+    STRING label;
+    BOOLEAN ml := false;
+    BOOLEAN mr := false;
+    INTEGER1 vr := 0;
+END;
+
+childlhs := SORTED(DATASET([{3, 'aaa', '', 0, ''}, {4, 'bbb', '', 0, ''}, {5, 'ccc', '', 0, ''}, {6, 'ddd', '', 0, ''}, {7, 'eee', '', 0, ''}], childrec), i);
+childrhs := SORTED(DATASET([{1, '', 'fff', 0, ''}, {3, '', 'ggg', 0, ''}, {5, '', 'hhh', 0, ''}, {5, '', 'iii', 0, ''}, {5, '', 'xxx', 0, ''}, {5, '', 'jjj', 0, ''}, {7, '', 'kkk', 0, ''}, {9, '', 'lll', 0, ''}, {9, '', 'mmm', 0, ''}], childrec), i);
+
+jrec := RECORD,MAXLENGTH(100)
+    UNSIGNED1 i;
+    STRING3 lstr;
+    STRING3 rstr;
+    UNSIGNED1 c;
+    STRING label;
+    dataset(childrec) children;
+    BOOLEAN ml := false;
+    BOOLEAN mr := false;
+    INTEGER1 vr := 0;
+END;
+
+lhs := SORTED(DATASET([{3, 'aaa', '', 0, '', childlhs}, {4, 'bbb', '', 0, '', childlhs}, {5, 'ccc', '', 0, '', childlhs}, {6, 'ddd', '', 0, '', childlhs}, {7, 'eee', '', 0, '', childlhs}], jrec), i);
+rhs := SORTED(DATASET([{1, '', 'fff', 0, '', childrhs}, {3, '', 'ggg', 0, '', childrhs}, {5, '', 'hhh', 0, '', childrhs}, {5, '', 'iii', 0, '', childrhs}, {5, '', 'xxx', 0, '', childrhs}, {5, '', 'jjj', 0, '', childrhs}, {7, '', 'kkk', 0, '', childrhs}, {9, '', 'lll', 0, '', childrhs}, {9, '', 'mmm', 0, '', childrhs}], jrec), i);
+
+trueval := true : stored('trueval');
+falseval := false : stored('falseval');
+
+BOOLEAN childmatch1(childrec l, childrec r) := (l.i = r.i);
+BOOLEAN childmatch2(childrec l, childrec r) := (r.rstr < 'x');
+BOOLEAN childmatch(childrec l, childrec r) := (childmatch1(l, r) AND childmatch2(l, r));
+
+BOOLEAN match1(jrec l, jrec r) := (l.i = r.i);
+BOOLEAN match2(jrec l, jrec r) := (r.rstr < 'x');
+BOOLEAN match(jrec l, jrec r) := (match1(l, r) AND match2(l, r));
+
+// transform for joins and non-group denormalizes, to be used with match
+childrec xfmchild(jrec lparent, childrec l, jrec rparent, childrec r, STRING lab) := TRANSFORM
+    SELF.i := l.i;
+    SELF.lstr := l.lstr;
+    SELF.rstr := r.rstr;
+    SELF.c := l.c+1;
+    SELF.label := lab;
+    SELF.ml := MATCHED(l);
+    SELF.mr := MATCHED(r);
+    SELF.vr := IF (MATCHED(r), -1, 1);
+END;
```

Comment:
You should also check the results of MATCHED(lParent), MATCHED(rParent) to ensure they are correct.

Although highly unusual things like this do occur.  A typical situation is where an attribute is defined in a transform based on MATCHED(LEFT).  That value is then passed into a function which happens to contain a join which is done as a child query, and the value that was passed in is used in that function's transform.